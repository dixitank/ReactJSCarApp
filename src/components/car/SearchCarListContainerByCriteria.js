import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import toastr from 'toastr';
import * as carAction from '../../action/CarAction';
import CarList from './CarList';



export class SearchCarListContainerByCriteria extends React.Component {

    constructor() {
        super();

        this.handleAddCar = this.handleAddCar.bind(this);
        this.handleSeachByMileage = this.handleSeachByMileage.bind(this);
    }


    componentDidMount() {
       
    }

    handleAddCar() {
        this.props.history.push('/addCar');
    }

    handleSeachByMileage() {
        this.props.history.push('/carsByMileage');
    }

    render() {
        const { cars } = this.props;

        if (!cars) {
            return (
                <div>Loading...</div>
            );
        }

        return (
            <div className="container-fluid">
                <div className="row mt-3">
                    <div className="col">
                        <h1>Cars</h1>                        
                    </div>
                </div>

                <div className="row mt-3">
                    <div className="col">
                        <div className="btn-group" role="group">
                            <button
                                type="button"
                                className="btn btn-primary"
                                onClick={this.handleAddCar}
                            >
                                <i className="fa fa-plus" aria-hidden="true"/> New
                            </button>

                            <button
                                type="button"
                                className="btn btn-warning ml-2"
                                onClick={this.handleSeachByMileage}
                            >
                                <i className="fa fa-plus" aria-hidden="true"/> Search By Mileage
                            </button>

                        </div>
                    </div>
                </div>

                <div className="row">
                    <div className="col">
                        <CarList cars={cars} />
                    </div>
                </div>
            </div>
        );
    }
}



const mapStateToProps = state => ({
    cars: state.carsReducer.cars
});



const mapDispatchToProps = dispatch => ({
    action: bindActionCreators(carAction, dispatch)

});



SearchCarListContainerByCriteria.propTypes = {
    cars: PropTypes.array,
    action: PropTypes.object.isRequired,
    history: PropTypes.object.isRequired
};



export default connect(mapStateToProps, mapDispatchToProps)(SearchCarListContainerByCriteria);
