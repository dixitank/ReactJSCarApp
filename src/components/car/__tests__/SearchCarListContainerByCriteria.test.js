import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {SearchCarListContainerByCriteria} from '../SearchCarListContainerByCriteria';


describe('SearchCarListContainerByCriteria.test.js', () => {

    it('renders without crashing', () => {
        const props = {
            action: {
                getCarAction: jest.fn()
            },
            cars: [{brand: 'Ford'}]
           
        };

        const wrapper = shallow(<SearchCarListContainerByCriteria {...props}/>);
        expect(wrapper.length).toEqual(1);

        const tree = toJson(wrapper);
        expect(tree).toMatchSnapshot();
    });

});



    
        