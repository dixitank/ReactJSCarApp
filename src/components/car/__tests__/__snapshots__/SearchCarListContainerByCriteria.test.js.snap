exports[`SearchCarListContainerByCriteria.test.js renders without crashing 1`] = `
<div
  className="container-fluid">
  <div
    className="row mt-3">
    <div
      className="col">
      <h1>
        Cars
      </h1>
    </div>
  </div>
  <div
    className="row mt-3">
    <div
      className="col">
      <div
        className="btn-group"
        role="group">
        <button
          className="btn btn-primary"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-plus" />
           New
        </button>
        <button
          className="btn btn-warning ml-2"
          onClick={[Function]}
          type="button">
          <i
            aria-hidden="true"
            className="fa fa-plus" />
           Search By Mileage
        </button>
      </div>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col">
      <CarList
        cars={
          Array [
            Object {
              "brand": "Ford",
            },
          ]
        } />
    </div>
  </div>
</div>
`;
